File: src/domain/entities/Organization.ts
interface Organization {
    id: string;
    name: string;
    email: string;
    projectId: string;
    privateKey: string;
    createdAt: string;
    updatedAt: string;
    username: string;
    enabled: boolean;
    accountNonExpired: boolean;
    credentialsNonExpired: boolean;
    accountNonLocked: boolean;
}

interface AuthenticateOrganizationDto {
    email: string
    password: string
}

interface AuthResponse {
    data: Organization;
    token: string;
}
interface CreateOrganizationDto {
    name: string
    email: string
    password: string
}

interface AuthenticationResourceOrganization {
    data: Organization;
    token: string;
}

File: src/main/java/inc/yowyob/service/snappy/domain/entities/Organization.java
package inc.yowyob.service.snappy.domain.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import inc.yowyob.service.snappy.infrastructure.helpers.LocalDateTimeDeserializer;
import inc.yowyob.service.snappy.infrastructure.helpers.LocalDateTimeSerializer;
import java.time.LocalDateTime;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Table("organizations")
public class Organization {

  @Id
  private UUID id;

  private String name;
  private String email;

  @JsonIgnore 
  private String password;

  @Column("project_id")
  private String projectId;

  @Column("private_key")
  private String privateKey;

  // Note: R2DBC doesn't support @OneToMany relationships
  // These would need to be managed through separate queries
  // private List<User> users;
  // private List<Chatbot> chatbots;

  @CreatedDate
  @JsonSerialize(using = LocalDateTimeSerializer.class)
  @JsonDeserialize(using = LocalDateTimeDeserializer.class)
  @Column("created_at")
  private LocalDateTime createdAt;

  @LastModifiedDate
  @JsonSerialize(using = LocalDateTimeSerializer.class)
  @JsonDeserialize(using = LocalDateTimeDeserializer.class)
  @Column("updated_at")
  private LocalDateTime updatedAt;

  public Organization(String name, String email, String password) {
    this.name = name;
    this.email = email;
    this.password = password;
  }
}
